src/test/org/codehaus/groovy/transform/ImmutableTransformTest.groovy

```diff
@@ -18,12 +18,40 @@
  */
 package org.codehaus.groovy.transform
 
+import org.codehaus.groovy.control.MultipleCompilationErrorsException
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestName
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import static org.junit.Assume.assumeTrue
+
 /**
- * @author Paul King
- * @author Tim Yates
+ * Tests for the @Immutable transform.
  */
+@RunWith(JUnit4)
 class ImmutableTransformTest extends GroovyShellTestCase {
 
+    @Rule public TestName nameRule = new TestName()
+
+    @Before
+    void setUp() {
+        super.setUp()
+        // check java version requirements
+        def m = System.getProperty("java.version") =~ /\d\.\d+/

```

Comment:
This may fail on jdk9 with the new versioning scheme.  Maybe the following would work since (at the moment) this returns 9 on jdk9 ea's.

```
new BigDecimal(System.getProperty('java.specification.version')) >= 1.8
```